### PR TITLE
feat: add fields with logx methods, support using third party logging libs.

### PR DESCRIPTION
**Pull request by kevwan**
Examples:

```go
logx.Infow("fetch URL",
	logx.Field("url", "http://localhost:8080/hello"),
	logx.Field("attempt", 3),
	logx.Field("backoff", time.Second),
)
logx.Errorw("failed to fetch URL",
	logx.Field("url", "http://localhost:8080/hello"),
	logx.Field("attempt", 3),
	logx.Field("backoff", time.Second),
)
logx.Sloww("slow to fetch URL",
	logx.Field("url", "http://localhost:8080/hello"),
	logx.Field("attempt", 3),
	logx.Field("backoff", time.Second),
)
logx.Error("hello")
logx.Infov(map[string]interface{}{
	"url":     "localhost:8080/hello",
	"attempt": 3,
	"backoff": time.Second,
})
logx.WithDuration(1100*time.Microsecond).Infow("slow to fetch URL",
	logx.Field("url", "localhost:8080/hello"),
	logx.Field("attempt", 3),
	logx.Field("backoff", time.Second),
)
logx.WithContext(context.Background()).WithDuration(1100*time.Microsecond).Errorw(
	"failed to fetch URL",
	logx.Field("url", "localhost:8080/hello"),
	logx.Field("attempt", 3),
	logx.Field("backoff", time.Second),
)
logx.WithDuration(1100*time.Microsecond).WithContext(context.Background()).Errorw(
	"failed to fetch URL",
	logx.Field("url", "localhost:8080/hello"),
	logx.Field("attempt", 3),
	logx.Field("backoff", time.Second),
)
```

Outputs like:

```json
{"@timestamp":"2022-04-30T13:00:23.692+08:00","attempt":3,"backoff":"1s","caller":"main.go:63","content":"failed to fetch URL","level":"error","url":"http://localhost:8080/hello"}
{"@timestamp":"2022-04-30T13:00:23.692+08:00","attempt":3,"backoff":"1s","caller":"main.go:68","content":"slow to fetch URL","level":"slow","url":"http://localhost:8080/hello"}
{"@timestamp":"2022-04-30T13:00:23.692+08:00","caller":"main.go:73","content":"hello","level":"error"}
{"@timestamp":"2022-04-30T13:00:23.692+08:00","attempt":3,"backoff":"1s","caller":"main.go:58","content":"fetch URL","level":"info","url":"http://localhost:8080/hello"}
{"@timestamp":"2022-04-30T13:00:23.692+08:00","caller":"main.go:74","content":{"attempt":3,"backoff":1000000000,"url":"localhost:8080/hello"},"level":"info"}
{"@timestamp":"2022-04-30T13:00:23.692+08:00","attempt":3,"backoff":"1s","caller":"main.go:79","content":"slow to fetch URL","duration":"1.1ms","level":"info","url":"localhost:8080/hello"}
{"@timestamp":"2022-04-30T13:00:23.692+08:00","attempt":3,"backoff":"1s","caller":"main.go:84","content":"failed to fetch URL","duration":"1.1ms","level":"error","url":"localhost:8080/hello"}
{"@timestamp":"2022-04-30T13:00:23.692+08:00","attempt":3,"backoff":"1s","caller":"main.go:90","content":"failed to fetch URL","duration":"1.1ms","level":"error","url":"localhost:8080/hello"}
{"@timestamp":"2022-04-30T13:01:23.693+08:00","caller":"usage.go:61","content":"CPU: 0m, MEMORY: Alloc=2.1Mi, TotalAlloc=2.1Mi, Sys=14.5Mi, NumGC=0","level":"stat"}
```

Using third party libs for logging:

https://github.com/zeromicro/zero-contrib/tree/main/logx/zapx

https://github.com/zeromicro/zero-contrib/tree/main/logx/logrusx

### CODE REVIEW

**Comment by smithyj** [comment]
这个功能还挺有用的

**Review by zhoushuguang — APPROVED** [review]
+LGTM

**Review by fynxiu — APPROVED** [review]
LGTM

**Comment by nchtw164** [comment]
LGTM

File: core/logx/durationlogger.go [hunk 1]
```diff
@@ -2,75 +2,75 @@ package logx
 
 import (
 	"fmt"
-	"io"
-	"sync/atomic"
 	"time"
 
 	"github.com/zeromicro/go-zero/core/timex"
+	"golang.org/x/net/context"
```
Comment (re-dylan) [inline core/logx/durationlogger.go:8]: should be  "context"

Comment (kevwan) [reply core/logx/durationlogger.go]: good catch!

**Comment by dtest11** [comment]
can we support a config to turn off/turn on  print caller on log
```json
{"level":"info","ts":1655138524.0536172,"caller":"main.go:29","msg":"hello"}
```
the caller is expensive ,
```go
func getCaller(callDepth int) string {
	_, file, line, ok := runtime.Caller(callDepth)
	if !ok {
		return ""
	}

	return prettyCaller(file, line)
}
```

**Pull request merged**